README.md

```diff
@@ -1,64 +1,119 @@
 # Gucci
 
-A simple CLI templating tool written in golang. I created this because I wanted something that was more powerful than `envsubt` to use when templating files on the command line. I also used golang because I wanted a single binary.
+A simple CLI templating tool written in golang.
 
 [![GitHub version](https://badge.fury.io/gh/noqcks%2Fgucci.svg)](https://badge.fury.io/gh/noqcks%2Fgucci)
 [![License](https://img.shields.io/github/license/noqcks/gucci.svg)](https://github.com/noqcks/gucci/blob/master/LICENSE)
 [![Build Status](https://travis-ci.org/noqcks/gucci.svg?branch=master)](https://travis-ci.org/noqcks/gucci)
 
-# Get
+## Installation
 
-If you have go installed
+If you have `go` installed:
 
 ```
-go get github.com/noqcks/gucci
+$ go get github.com/noqcks/gucci
 ```
 
-Or you can just download the binary and move it into your path
+Or you can just download the binary and move it into your `PATH`:
 
 ```
-VERSION=0.0.4
+VERSION=0.1.0
 wget -q https://github.com/noqcks/gucci/releases/download/v${VERSION}/gucci-v${VERSION}-darwin-amd64
 chmod +x gucci-v${VERSION}-darwin-amd64
 mv gucci-v${VERSION}-darwin-amd64 /usr/local/bin/gucci
 ```
 
+## Use
+
+### Locating Templates
 
-# Use
+`gucci` can locate a template in multiple ways.
 
-gucci can take input in multiple ways
+#### File
 
-### file
+Pass the template file path as the first argument:
 
 ```
-$ gucci template.tpl > template.conf
+$ gucci template.tpl > template.out
 ```
 
-### stdin
+#### Stdin
 
-```
+Supply the template through standard input:
+
+```bash
 $ gucci
 Start typing stuff {{ print "here" }}
 ^d
 Start typing stuff here
 ```
 
-via piping
+Via piping:
 
-```
+```bash
 $ echo '{{ html "<escape-me/>" }}' | gucci
 ```
 
-# Templating
+### Supplying Variable Inputs
+
+`gucci` can receive variables for use in templates in the following ways (in order of lowest to highest precedence):
+* A JSON or YAML file
+* Environment variables
+* Variable command options
+
+#### Variables File
+
+Given an example variables file:
+```yaml
+# vars.yaml
+hosts:
+  - name: bastion
+  - name: app
+```
+
+Pass it into `gucci` with `-f` or `--vars-file`:
+```bash
+$ gucci -f vars.yaml template.tpl
+```
+
+#### Environment Variables
+
+Here, `MY_HOST` is available to the template:
+```bash
+$ export MY_HOST=localhost
+$ gucci template.tpl
+```
+
+#### Variable Options
+
+Pass variable options into `gucci` with `-s` or `--set-var`, which can be repeated:
+```bash
+$ gucci -s foo.bar=baz template.tpl
```

Comment:
Is there a way to pass arrays via -s?

Reply:
Hmm. Good idea. Not currently. Lmk if that's a good future enhancement or if you want me to take a stab at that in this PR.

Reply:
Let's put it as an GH issue or TODO item. Don't wanna increase scope of this PR anymore.

Reply:
https://github.com/noqcks/gucci/issues/13